### ISSUE TITLE
factory: build fault-recovery, energy-preference, and MoE-router distillation datasets

### ISSUE THREAD

**Issue by rmems** [rmems/synthetic-factory#78]
## Planning / research attribution

**GPT-5.6-Sol (OpenAI / ChatGPT), 2026-08-23**, with Raul Montoya Cardenas directing the multi-generator neuromorphic dataset program.

## Parent / scope

Parent epic: #76.

Implement three control-oriented neuromorphic dataset families:

- `neuromorphic-fault-recovery`
- `snn-energy-routing-preferences`
- `moe-router-distillation-trajectories`

These are directly relevant to the future Spikenaut artificial-nervous-system research direction, but they must be built and validated before any live LLM↔SNN coupling.

## neuromorphic-fault-recovery

Generator creates disturbances such as:

```text
sensor loss
stale sensor
event jitter
burst corruption
thermal excursion
missing channel
malformed spike burst
delayed result
temporary saturation
```

Oracle: relay + simulator and, where appropriate, real hardware replay.

Expected outcomes should distinguish:

```text
continue
degrade gracefully
fallback
reflex action
quarantine
fail closed
```

## snn-energy-routing-preferences

This family must use **measured runs**.

A generator may create equivalent control/task alternatives, but it may not invent their energy labels.

Example:

```text
policy A → correct → 31 J
policy B → correct → 8 J
policy C → wrong   → 0.2 J
```

The preferred policy is not simply the lowest-energy policy.

Objective:

> minimize energy subject to task-quality and safety constraints.

Store, where measurable:

```text
state
candidate action
task quality
latency
joules
thermal response
success/failure
preference
```

## moe-router-distillation-trajectories

Generator supplies diverse text/code/task contexts.

Teacher/oracle is the **real LLM/MoE router**.

Capture compact targets such as:

```text
top-k experts
router logits where available
top-1/top-2 margin
routing entropy
expert agreement
```

Do not ask GPT-5.6-Sol or another generator to invent what a different model's router would have selected.

## Baseline requirement

Before using an SNN student for router distillation, test at least one simple conventional baseline:

- linear model, and/or
- small MLP

If the target is not learnable from the proposed compact inputs, do not escalate to an SNN merely for novelty.

## Relationships

- **Parent:** #76
- **Blocked by / authority contract:** #32
- **Related raster/gate work:** #12
- **Milestone:** Factory v0.2.0
- **Future consumers:** standalone Spikenaut experiments first; live LLM↔SNN research only after independent baselines exist

## Acceptance criteria

- [ ] Fault scenarios have explicit recovery/fail-closed outcomes.
- [ ] Energy labels come from measured executions.
- [ ] Energy preference incorporates task-quality and safety constraints.
- [ ] MoE labels come from the actual teacher router.
- [ ] Teacher model/checkpoint/configuration are recorded.
- [ ] A trivial baseline is evaluated before SNN router distillation.
- [ ] No dataset claims SNN efficiency from theoretical operation counts alone.
- [ ] Generator-proposed scenario fields remain distinguishable from oracle-measured result fields.

## Tracking / Linear sync

GitHub is the canonical repo-scoped issue. The existing GitHub→Linear integration should create the Linear twin automatically; do **not** manually create a duplicate twin. After the twin appears, mirror the existing project/milestone/relationship metadata there because those fields are not guaranteed to sync.

**Comment by rmems** [comment]
**DELETE** — recommendation from **01 Synthetic Factory Debt Closer** (Tech Debt Mill).

Debt-mode freeze: no new train/gen, VSET–SWE campaign, or neuromorphic dataset-family work. Close unless Raul explicitly reopens for a named mill.

**Comment by rmems** [comment]
**Status update 2026-09-04**

Open: PR #138 (head `d20f6f37`) merges clean against `main` and is the first of the three oracle PRs to land once the #172 foundation exists; its generic envelope pieces move to `pipelines/oracle_grounded/envelope.py` and the distillation-specific rest to `distill_contract.py`. The energy meter and real MoE teacher runs remain external work as the PR states.

— Claude Fable 5.1 (Claude Code)
